File: src/windows/AmbientContextMcp.Core.Tests/StateMetadataTests.cs
using AmbientContextMcp.Core.Hub;
using AmbientContextMcp.Core.Models;
using Xunit;

namespace AmbientContextMcp.Core.Tests;

public class StateMetadataTests
{
    [Fact]
    public void GetContextStates_omits_metadata_when_requested()
    {
        var hub = LocalContextHubTestFactory.CreateInMemory();
        var observedAt = DateTimeOffset.UtcNow;

        hub.Ingest(new AmbientContextSnapshot
        {
            ObservedAt = observedAt,
            OutboundStates =
            [
                new()
                {
                    ObservedAt = observedAt,
                    Name = "battery.bucket",
                    Value = "ok",
                    Sensitivity = "low"
                }
            ]
        });

        var response = hub.GetContextStates(new LocalContextStateRequest
        {
            IncludeMetadata = false
        });

        var state = Assert.Single(response.States);
        Assert.Equal("battery.bucket", state.Name);
        Assert.Equal("ok", state.Value);
        Assert.Null(state.ObservedAt);
        Assert.Null(state.Sensitivity);
    }
}

File: src/macos/Tests/AmbientContextCoreTests/StateMetadataTests.swift
import Foundation
import Testing
@testable import AmbientContextCore

@Suite("StateMetadata")
struct StateMetadataTests {
    @Test("GetContextStates_omits_metadata_when_requested")
    func getContextStatesOmitsMetadataWhenRequested() throws {
        let hub = LocalContextHubTestFactory.createInMemory()
        let observedAt = Date()

        hub.ingest(AmbientContextSnapshot(
            observedAt: observedAt,
            outboundStates: [
                AmbientState(
                    observedAt: observedAt,
                    name: "battery.bucket",
                    value: "ok",
                    sensitivity: "low")
            ]))

        let response = hub.getContextStates(LocalContextStateRequest(includeMetadata: false))

        #expect(response.states.count == 1)
        let state = try #require(response.states.first)
        #expect(state.name == "battery.bucket")
        #expect(state.value == "ok")
        #expect(state.observedAt == nil)
        #expect(state.sensitivity == nil)
    }
}
